contracts/std-lib-extended/src/result.sw

```diff
@@ -0,0 +1,39 @@
+library result;
+
+struct ResultErrRevert<M, E> {
+    message: M,
+    error: E,
+}
+
+impl<T, E> Result<T, E> {
```

Comment:
ditto https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/result.sw#L4 ?

Reply:
Same reasoning as https://github.com/hyperlane-xyz/fuel-contracts/pull/52#discussion_r1152125562

Also feels crazy to me that their unwrap doesn't bubble up the error at all https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/result.sw#L149